std/algorithm/iteration.d

```diff
@@ -2868,6 +2871,359 @@ private template ReduceSeedType(E)
     }
 }
 
+/++
+Similar to $(LREF reduce), but returns a range containing the
+successive reduced values. The call $(D scan!(fun)(range, seed))
+first assigns $(D seed) to an internal variable $(D result),
+also called the accumulator. The returned range contains the values
+$(D result = fun(result, x)) lazily evaluated for each element $(D x)
+in $(D range). Finally, the last element has the same value as
+$(D reduce!(fun)(seed, range)).
+The one-argument version $(D scan!(fun)(range)) works similarly, but
+it returns the first element unchanged and uses it as seed for the
+next elements.
+This function is also known as $(D partial_sum), $(D accumulate),
+$(D prefix_sum), or $(D cumulative_sum).
+
+Returns:
+    a range containg the consecutive reduced values. If there is more than one
+    fun, the element type will be $(D Tuple) containing one element for each fun.
```

Comment:
Nitpick, but probably sensible to link `Tuple`, e.g. with `$(REF Tuple, std, typecons)`.